### PR TITLE
refactor: add better message delete event handlers

### PR DESCRIPTION
**Pull request by kyranet**
| ⚠️ This Pull Request needs https://github.com/skyra-project/klasa/pull/11 to be approved and merged beforehand. Once the linked Pull Request is merged, this one will receive its last commit to re-target to the fork's `settings` branch. |
| --- |

### CODE REVIEW

File: src/events/messages/messageDeleteBulkRemoveResponses.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { Events } from '#lib/types/Enums';
+import { ApplyOptions } from '@skyra/decorators';
+import { Collection, Message, Snowflake } from 'discord.js';
+import { Event, EventOptions } from 'klasa';
+
+@ApplyOptions<EventOptions>({ event: Events.MessageDeleteBulk })
+export default class extends Event {
+	public async run(messages: Collection<Snowflake, Message>) {
+		for (const message of messages.values()) {
+			for (const response of message.responses) {
```
Comment (favna) [inline src/events/messages/messageDeleteBulkRemoveResponses.ts:10]: I think it's better to emit messageDelete here we we follow DRY pattern

**Review by favna — CHANGES_REQUESTED** [review]

File: src/events/messages/messageDeleteBulkRemoveResponses.ts (hunk 1 above)
Comment (kyranet) [reply src/events/messages/messageDeleteBulkRemoveResponses.ts:10]: Do you know how many things you break in discord.js from doing this?

Comment (favna) [reply src/events/messages/messageDeleteBulkRemoveResponses.ts:10]: Wdym? Just replace this inner for loop with `this.client.emit('messageDelete', message)`

Comment (kyranet) [reply src/events/messages/messageDeleteBulkRemoveResponses.ts:10]: `messageDelete` is a discord.js event.

**Review by favna — DISMISSED** [review]

**Pull request merged**